modules/swagger-codegen/src/main/resources/csharp/ApiClient.mustache

```diff
@@ -340,26 +379,55 @@ namespace {{packageName}}.Client
                     }
                 }
                 var stream = new MemoryStream(response.RawBytes);
+                log.Trace("end");
                 return stream;
             }
 
-            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")) // return a datetime object
+            if (type.Name.StartsWith("System.Nullable`1[[System.DateTime")
             {
+                // return a datetime object 
+                log.Trace("end");
                 return DateTime.Parse(response.Content,  null, System.Globalization.DateTimeStyles.RoundtripKind);
             }
 
-            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")) // return primitive type
+            if (type == typeof(String) || type.Name.StartsWith("System.Nullable")
             {
+                // return primitive type 
+                log.Trace("end");
                 return ConvertType(response.Content, type);
             }
 
+            log.Trace("end");
+            return DeserializeObject(response, type);
+        }
+
+        /// <summary>
+        /// Deserialize the JSON string into a proper object.
+        /// </summary>
+        /// <typeparam name="T">Model object</typeparam>
+        /// <param name="response">The HTTP response.</param>
+        /// <returns>Object representation of the JSON string.</returns>
+        public T Deserialize<T>(IRestResponse response
+        {
+            log.Trace("start");
+            var type = typeof(T);
             // at this point, it must be a model (json)
-            try
+            if (!(DeserializeObject(response, type) is T)
             {
-                return JsonConvert.DeserializeObject(response.Content, type, serializerSettings);
+                return default(T);
```

Comment:
To be clear, this isn't standard behavior, because if my code is supposed to deserialize `Cat`, but it attempts to deserialize `Dog` then your new code is swallowing an obvious issue. We'd want to either return a 500 message here (meaning the server screwed up somewhere), or possibly a 400 Bad Request.  I don't think we can easily assume response code 400 for this scenario, so we should probably stick with 500.